lib/dock.rb

```diff
@@ -0,0 +1,77 @@
+class Dock
+  attr_reader :name, :max_rental_time, :boats_rented
+
+  def initialize(name, max_rental_time)
+    @name = name
+    @max_rental_time = max_rental_time
+    @boats_rented = []
+  end
+
+  def rent(boat, renter)
+    @boats_rented << boat
+  end
+
+  def return(boat)
+    @boats_rented.delete(boat)
+  end
+
+  def log_hour
+    @boats_rented
+    @boats_rented.each do |boat|
+      boat.add_hour
+      boat.hours_rented
```

Comment:
Not sure if we need to call `hours_rented`? Need to increase the hours for each boat, but after that think that `log_hour` has pretty much done its initial job.